### PR TITLE
Breadth-first order for crawling

### PR DESCRIPTION
**Pull request by ivanprado**
BF order for crawling is better suited IMO for autoextract-spiders

Also, I included a trick to force the crawling of the seeds. Otherwise, some of them can get ignored. 

This PR depends on https://github.com/scrapinghub/autoextract-spiders/pull/25

### CODE REVIEW

File: autoextract_spiders/spiders/crawler_spider.py [hunk 1]
```diff
@@ -223,11 +223,17 @@ def _schedule_seed_urls(self, seed_urls):
...
-            yield Request(url,
+            request = Request(url,
                           meta={'source_url': url},
                           callback=self.main_callback,
                           errback=self.main_errback,
                           dont_filter=True)
+            # Trick required to avoid some seeds to be never processed or too late.
+            try:
+                self.crawler.engine.crawl(request, self)
+            except AssertionError:
```
Comment (vshlapakov) [inline autoextract_spiders/spiders/crawler_spider.py:234]: Out of curiosity, what are the conditions to trigger the assertion error?

**Review by vshlapakov — APPROVED** [review]
Looks good to me, that should work better in general imho.

File: autoextract_spiders/spiders/crawler_spider.py (hunk 1 above)
Comment (ivanprado) [reply autoextract_spiders/spiders/crawler_spider.py:234]: To be honest, no idea. I copied this fix to get all initial URLs fetched form other projects. @kmike  do you know which conditions raise this assertion error?

Comment (kmike) [reply autoextract_spiders/spiders/crawler_spider.py:234]: I recall it fails always if you remove this condition; something about engine not started, for a first request(s), but I'm not sure.

Comment (vshlapakov) [reply autoextract_spiders/spiders/crawler_spider.py:234]: Okay, that makes some sense, thanks guys

**Pull request merged**